src/fs.cpp

```diff
@@ -148,7 +148,10 @@ std::string get_filesystem_error_message(const fs::filesystem_error& e)
 #ifdef __GLIBCXX__
 
 // reference: https://github.com/gcc-mirror/gcc/blob/gcc-7_3_0-release/libstdc%2B%2B-v3/include/std/fstream#L270
-
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wswitch"
```

Comment:
Is this commit " Fix Windows build with --enable-werror on Ubuntu Bionic (gcc 7.5) " still needed?

We no longer use mingw on bionic (only focal)

Reply:
Right. #21036 made it possible.

Going to rebase the pr and remove that commit.

Reply:
@MarcoFalke It still required to prevent `-Werror=switch` warnings on Focal.

Therefore, technically, no changes are required in the patch.